dev-libs/leveldb/leveldb-1.20.recipe

```diff
@@ -0,0 +1,80 @@
+SUMMARY="A fast key-value storage library written at Google"
+DESCRIPTION="LevelDB stores keys and values in arbitrary byte arrays, and \
+data is sorted by key. It supports batching writes, forward and backward \
+iteration, and compression of the data via Google's Snappy compression \
+library.
+LevelDB is not an SQL database. Like other NoSQL and Dbm stores, it does \
+not have a relational data model and it does not support SQL queries. Also, \
+it has no support for indexes. Applications use LevelDB as a library, as it \
+does not provide a server or command-line interface."
+HOMEPAGE="http://leveldb.org/"
+COPYRIGHT="2011 The LevelDB Authors"
+LICENSE="BSD (3-clause)"
+REVISION="1"
+SOURCE_URI="https://github.com/google/leveldb/archive/v$portVersion.tar.gz"
+CHECKSUM_SHA256="f5abe8b5b209c2f36560b75f32ce61412f39a2922f7045ae764a2c23335b6664"
+PATCHES="leveldb-$portVersion.patchset"
+
+ARCHITECTURES="!x86_gcc2 ?x86 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+libVersion="$portVersion"
+libVersionCompat="$libVersion compat >= ${libVersion%%.*}"
+
+PROVIDES="
+	leveldb$secondaryArchSuffix = $portVersion
+	lib:libleveldb$secondaryArchSuffix = $libVersionCompat
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libsnappy$secondaryArchSuffix
+	"
+
+PROVIDES_devel="
+	leveldb${secondaryArchSuffix}_devel = $portVersion
+	devel:libleveldb$secondaryArchSuffix = $libVersionCompat
+	devel:libmemenv$secondaryArchSuffix = $libVersionCompat
+	"
+REQUIRES_devel="
+	leveldb$secondaryArchSuffix == $portVersion base
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libsnappy$secondaryArchSuffix
+	"
+BUILD_PREREQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	cmd:gcc$secondaryArchSuffix
+	cmd:ld$secondaryArchSuffix
+	cmd:libtoolize$secondaryArchSuffix
+	cmd:make
+	"
+
+defineDebugInfoPackage leveldb$secondaryArchSuffix \
+	"$libDir"/libleveldb.so.$libVersionCompat
```

Comment:
You really have to change `"$libDir"/libleveldb.so.$libVersionCompat` to `"$libDir"/libleveldb.so.$libVersion` and set the right value for `libVersionCompat`, because `compat >= 1` is _not_ part of the filename.

You could also revert to `"$libDir"/libleveldb.so.1`, but you would still have to set the right value for `libVersionCompat`, otherwise your recipe will report an incorrect version.

Reply:
This is puzzling: According to archlinux, the name of the lib is not `libleveldb.so.1` but `libleveldb.so.1.20`, so I don't understand why you got an error when you had `libVersion="1.20"`.
Source: https://www.archlinux.org/packages/extra/x86_64/leveldb/files/